lib/TAP/Harness.pm

```diff
@@ -555,8 +555,10 @@ sub runtests {
         $self->_make_callback( 'after_runtests', $aggregate );
     };
     my $run = sub {
-        $self->aggregate_tests( $aggregate, @tests );
-        $finish->();
+        my $bailout;
+        eval { $self->aggregate_tests( $aggregate, @tests ); 1 }
+            or do { $bailout = $@ || 'unknown_error' };
+        $finish->( defined $bailout );
```

Comment:
Can something more useful not be done with `$bailout` than just use it as a truth value?

Reply:
Maybe, but I consider it out of scope of this change.

Reply:
It would require changing at least the `summary` meaning, maybe even some change to `$aggregate`, which implies changes in dependencies. I don't see this happening in foreseeable future.

Reply:
You are changing the semantic behaviour of a very important piece of code more than absolutely necessary. I believe it to absolutely be in scope.

I think a better way forward is to leave the call to `$finish` exactly as before, but afterwards to rethrow any exception (ie `$bailout`) if it's defined.

Reply:
Ok, sounds reasonable.

Reply:
Fixed